crates/storage/db/src/tables/mod.rs

```diff
@@ -176,14 +173,23 @@ table!(
 
 table!(
     /// Stores the mapping of block number to state transition id.
-    ( BlockTransitionIndex ) BlockNumHash | TransitionId
+    /// The block transition marks the final state at the end of the block.
+    /// Increment the transition if the block contains an addition block reward.
+    /// If the block does not have a reward and transaction, the transition will be the same as the
+    /// transition at the last transaction of this block.
+    ( BlockTransitionIndex ) BlockNumber | TransitionId
```

Comment:
Would prefer we do this change in a standalone PR, probably needs to be part of our broader change to only using block numbers? @onbjerg @joshieDo

Reply:
Asked in chat, but will move it to separate PR for a cleaner look. It is already changed everywhere it is used.

Reply:
Split the change to separate PR here: https://github.com/paradigmxyz/reth/pull/894